projects/igniteui-angular/src/lib/core/styles/components/carousel/_carousel-theme.scss

```diff
@@ -87,7 +87,9 @@
     }
 
     @if not($button-disabled-arrow-color) and $button-disabled-background {
-        $button-disabled-arrow-color: text-contrast($button-disabled-background);
+        @if type_of($button-disabled-background) == 'color' {
```

Comment:
I don't think this check is necessary. If the user passes anything other than a color, the compiler would just scream at them that a color must be passed.